### PR TITLE
feat(steering_offset_estimator): add steering offset estimation package

### PR DESCRIPTION
**Pull request by taikitanaka3**
Signed-off-by: tanaka3 <ttatcoder@outlook.jp>

## Description

add steering offset estimation and diag for error because steer offset effects vehicle behavior very much currently.

## Related links

#929 

## Tests performed

by plot juggler + logsim +
`ros2 launch steer_offset_estimator steer_offset_estimtor.launch.xml vehicle_model:=lexus` or anything

image

## Notes for reviewers

need to remove steering offset estimation from internal calib repos

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: vehicle/steer_offset_estimator/src/steer_offset_estimator_node.cpp [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+// function for diagnostics
+void SteerOffsetEstimatorNode::monitorSteerOffset(DiagnosticStatusWrapper & stat)
+{
+  using DiagStatus = diagnostic_msgs::msg::DiagnosticStatus;
+  const double eps = 1e-3;
+  if (covariance_ > eps) {
+    stat.summary(DiagStatus::OK, "Preparation");
+    return;
+  }
+  if (estimated_steer_offset_ > warn_steer_offset_) {
```
Comment (shmpwk) [inline vehicle/steer_offset_estimator/src/steer_offset_estimator_node.cpp:74]:
```suggestion
  if (std::abs(estimated_steer_offset_) > warn_steer_offset_) {
```
I noticed that the word `warn_steer_offset_` is ambiguous in that it is absolute value or not.
This suggestion is on the premise of absolute value.

**Review by shmpwk — COMMENTED** [review]
1. `warn_steer_offset_`  is ambiguous in that it is absolute value or not.
If this is absolute value, `warn_abs_steer_offset_` is better?


2. This PR should be along with removing the duplicated package in calibration tool?

File: vehicle/steer_offset_estimator/src/steer_offset_estimator_node.cpp [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+: Node("steer_offset_estimator", node_options)
+{
+  using std::placeholders::_1;
+  // get parameter
+  const auto vehicle_info = vehicle_info_util::VehicleInfoUtil(*this).getVehicleInfo();
+  wheel_base_ = vehicle_info.wheel_base_m;
+  covariance_ = this->declare_parameter("initial_covariance", 1000.0);
+  forgetting_factor_ = this->declare_parameter("forgetting_factor", 0.999);
+  update_hz_ = this->declare_parameter<double>("steer_update_hz", 10.0);
+  valid_min_velocity_ = this->declare_parameter<double>("valid_min_velocity");
```
[the remark this answers is not in the document]
Comment (taikitanaka3) [reply vehicle/steer_offset_estimator/src/steer_offset_estimator_node.cpp:34]: @shmpwk I want this value set properly by product so I removed default value.

[the remark this answers is not in the document]
Comment (shmpwk) [reply vehicle/steer_offset_estimator/src/steer_offset_estimator_node.cpp:34]: I see!

**Comment by shmpwk** [comment]
ros2 launch steer_offset_estimator steer_offset_estimtor.launch.xml vehicle_model:=sample_vehicle

Screenshot from 2022-06-02 19-05-29

**Comment by taikitanaka3** [comment]
@tkimura4 
I think merge is ready.
TODO : add aggregator setting to autoware_launcher

**Comment by yukkysaito** [comment]
@taikitanaka3 Can you write the README?

**Comment by taikitanaka3** [comment]
@yukkysaito 
I have added readme.md

**Comment by yukkysaito** [comment]
@taikitanaka3 Thank you!
Pls add the following items to readme :pray: 
- Purpose
- Inner-workings / Algorithms
- Inputs / Outputs
- Parameters
Sample: 
https://github.com/autowarefoundation/autoware.universe/blob/main/perception/detection_by_tracker/README.md

**Comment by taikitanaka3** [comment]
@yukkysaito 
I added :+1:

**Comment by shmpwk** [comment]
This PR should sync merge https://github.com/tier4/calibration_tools.iv/pull/149 which has already merged.

**Comment by mitsudome-r** [comment]
@yukkysaito @shmpwk Is this PR ready for merge?

**Comment by shmpwk** [comment]
@mitsudome-r Yes ready

**Pull request merged**